packages/service-library/src/servicelib/rabbitmq/_utils.py

```diff
@@ -50,8 +60,26 @@ async def wait_till_rabbitmq_responsive(url: str) -> bool:
     return await is_rabbitmq_responsive(url)
 
 
+def _get_unique_rabbit_queue_name_prefix() -> str:
+    # NOTE: this prefix is guaranteed to be unique for the entire lifecycle of the docker container
+    # Why is this desiarable?
+    # 1. the code base makes the above assumption, otherwise subcscribers and consumers do not work
+    # 2. enables restaratability of production deployemnts, where containers are frist created and
+    # then killed; avoids overlapping queue names and errors during start
+    prefix: str | None = None
+    if _PATH_UNIQUE_RABBIT_QUEUE_PREFIX.exists():
+        prefix = _PATH_UNIQUE_RABBIT_QUEUE_PREFIX.read_text()
+
+    if prefix is None:
+        random_str = "".join(random.choices(_ALPHABET, k=_CHAR_COUNT))  # noqa: S311
+        prefix = f"{socket.gethostname()}_{random_str}"
+        _PATH_UNIQUE_RABBIT_QUEUE_PREFIX.write_text(prefix)
+
+    return prefix
+
+
 def get_rabbitmq_client_unique_name(base_name: str) -> str:
-    return f"{base_name}_{socket.gethostname()}"
+    return f"{base_name}_{_get_unique_rabbit_queue_name_prefix()}"
```

Comment:
Why are we using the client name for queue names??? this seems very wrong.

Reply:
discussed